src/auth/s2s/s2s.class.ts

```diff
@@ -0,0 +1,95 @@
+import events from 'events'
+import { NextFunction, Request, RequestHandler, Response, Router } from 'express'
+// eslint-disable-next-line @typescript-eslint/camelcase
+import jwt_decode from 'jwt-decode'
+import { DecodedJWT } from './decodedJwt.interface'
+import { S2S } from './s2s.constants'
+import { S2SConfig } from './s2sConfig.interface'
+import { S2SToken } from './s2sToken.interface'
+import { postS2SLease } from './serviceAuth'
+
+// TODO: To be replaced with a proper logging library
+const logger = console
+
+// Cache of S2S tokens, indexed by microservice name
+const cache: { [key: string]: S2SToken } = {}
+
+export class S2SAuth extends events.EventEmitter {
+    protected readonly router = Router({ mergeParams: true })
+
+    protected s2sConfig!: S2SConfig
```

Comment:
Why is this marked as not null (with the exclamation mark), but isn't it given a default value?

Reply:
This has now been given a default, dummy config